### PR TITLE
 feat(templates): makes ssl_session_cache configurable

### LINKED ISSUE #9994: Cannot override ssl_session_cache using nginx directives
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.2.2

### Current Behavior

When I use the ENV variable `KONG_NGINX_PROXY_SSL_SESSION_CACHE` to set `ssl_session_cache` in the nginx config of the proxy Kong won't start and fail with the error message:

```
nginx: [emerg] the size 52428800 of shared memory zone "SSL" conflicts with already declared size 10485760 in /kong_prefix/nginx-kong.conf:140
```

### Expected Behavior

The variable `KONG_NGINX_PROXY_SSL_SESSION_CACHE` should override `ssl_session_cache` in the nginx config as described in https://docs.konghq.com/gateway/2.8.x/reference/configuration/#injecting-individual-nginx-directives

### Steps To Reproduce

```markdown
docker run --name kong -e "KONG_NGINX_PROXY_SSL_SESSION_CACHE=shared:SSL:50m" \
    -e "KONG_DATABASE=off" \
    -e "KONG_PROXY_ACCESS_LOG=/dev/stdout" \
    -e "KONG_ADMIN_ACCESS_LOG=/dev/stdout" \
    -e "KONG_PROXY_ERROR_LOG=/dev/stderr" \
    -e "KONG_ADMIN_ERROR_LOG=/dev/stderr" \
    -e "KONG_ADMIN_LISTEN=0.0.0.0:8001, 0.0.0.0:8444 ssl" \
    -p 8000:8000 \
    -p 8443:8443 \
    -p 8001:8001 \
    -p 8444:8444 \
    kong:2.8

Error: could not prepare Kong prefix at /usr/local/kong: nginx configuration is invalid (exit code 1):
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /usr/local/kong/nginx.conf:6
nginx: [emerg] the size 52428800 of shared memory zone "SSL" conflicts with already declared size 10485760 in /usr/local/kong/nginx-kong.conf:130
nginx: configuration file /usr/local/kong/nginx.conf test failed
```

### Anything else?

This change would be my fix for https://support.konghq.com/support/s/article/Could-not-allocate-new-session-in-SSL-session-shared-cache

### PR DESCRIPTION
**Pull request by michbeck100**
### Summary

This change makes ssl_session_cache configurable in nginx-kong.conf and nginx-kong-stream.conf.

### Issue reference

Fix #9994

### CODE REVIEW

File: kong.conf.default
Comment (chronolaw) [inline kong.conf.default:718]: I think this item should be renamed as `ssl_session_cache_size`.

File: spec/fixtures/custom_nginx.template
Comment (chronolaw) [inline spec/fixtures/custom_nginx.template:111]: We should also change `shared:StreamSSL:10m;` in this file.

**Review by chronolaw — APPROVED** [review]
LGTM, thank you @michbeck100 .

**Comment by chronolaw** [comment]
Could you add a change log entry for this PR?

Here is a reference PR: #9624 .

**Comment by michbeck100** [comment]
> Could you add a change log entry for this PR?
> 
> Here is a reference PR: #9624 .

Done

**Review by dndx — APPROVED** [review]

**Comment by dndx** [comment]
@michbeck100 Thanks for your contributions.

Could you do a quick rebase onto `master`? It appears I do not have permission to update your branch because "allow project owners to edit PR" is not turned on. I will merge this PR once the merge conflict is resolved.

Thanks!

**Comment by michbeck100** [comment]
> @michbeck100 Thanks for your contributions.
> 
> Could you do a quick rebase onto `master`? It appears I do not have permission to update your branch because "allow project owners to edit PR" is not turned on. I will merge this PR once the merge conflict is resolved.
> 
> Thanks!

Done

File: CHANGELOG.md
Comment (chronolaw) [inline CHANGELOG.md:91]: `nginx` => `Nginx`

**Review by dndx — APPROVED** [review]

**Comment by dndx** [comment]
Merged. Thanks for your contribution @michbeck100 !

**Pull request merged**
